firmware/application/ui/ui_receiver.cpp

```diff
@@ -59,48 +63,75 @@ void FrequencyField::set_value(rf::Frequency new_value) {
 }
 
 void FrequencyField::set_step(rf::Frequency new_value) {
-    step = new_value;
     // TODO: Quantize current frequency to a step of the new size?
+    step_ = new_value;
 }
 
 void FrequencyField::paint(Painter& painter) {
-    const std::string str_value = to_string_short_freq(value_);
-
+    const auto str_value = to_string_short_freq(value_);
     const auto paint_style = has_focus() ? style().invert() : style();
 
     painter.draw_string(
         screen_pos(),
         paint_style,
         str_value);
+
+    // Highlight current digit in digit_mode.
+    if (digit_mode_) {
+        auto p = screen_pos();
+        p += {digit_ * char_width, 0};
+        painter.draw_char(p, Styles::bg_blue, str_value[digit_]);
+    }
 }
 
-bool FrequencyField::on_key(const ui::KeyEvent event) {
-    if (event == ui::KeyEvent::Select) {
+bool FrequencyField::on_key(KeyEvent event) {
+    if (event == KeyEvent::Select) {
+        // Toggle 'digit' mode with long-press.
+        if (digit_mode_ || key_is_long_pressed(event)) {
+            digit_mode_ = !digit_mode_;
+            set_dirty();
+            return true;
+        }
+
         if (on_edit) {
             on_edit();
             return true;
         }
     }
+
+    if (digit_mode_) {
+        switch (event) {
+            case KeyEvent::Up:
+                set_value(value_ + digit_step());
+                break;
+            case KeyEvent::Down:
+                set_value(value_ - digit_step());
+                break;
+            case KeyEvent::Left:
+                digit_--;
+                break;
+            case KeyEvent::Right:
```

Comment:
If you wanted to get fancy here, you could skip over the decimal point, versus letting me try to change its value with the encoder dial.  :-)

Reply:
I thought about it. Easy enough to add the skip over if we want.

Reply:
One thing I really haven’t decided on is whether a change that would put it out of range should just be ignored vs the current behavior where it ends up clamped. Might be nicer to have it not lose the rest of the digit values in that case.

Reply:
If you ignored out of range frequencies, would the previous in-range frequency still be programmed into the receiver, or would you disable the receiver and change the field to red or something?  I think it would be undesirable if the display showed 9999.999 while I was still listening to 97.9.  :-)

Reply:
I’d just prevent it from being updated to an out of range value in the first place. So if you used the encoded it would just stop updating if the next update is out of range.